### DISCUSSION TITLE
Invalid RBAC credentials

### DISCUSSION THREAD

**Discussion by marlon617** [Kong/kong#8388] [Help] [answered]
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/5423846/153172986-2dd62375-0f6b-480a-a9c3-b10e176a433e.png">

Reset RBAC Token后，重启pod  报错Invalid RBAC credentials

time="2022-02-09T09:14:38Z" level=info msg="diagnostics server disabled"
time="2022-02-09T09:14:38Z" level=info msg="starting controller manager" commit=119dca97739dafee5f0539b96774144754328e8d logger=setup release=2.1.1 repo="https://github.com/Kong/kubernetes-ingress-controller.git"
time="2022-02-09T09:14:38Z" level=info msg="getting enabled options and features" logger=setup
time="2022-02-09T09:14:38Z" level=info msg="getting the kubernetes client configuration" logger=setup
W0209 09:14:38.811052       1 client_config.go:615] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2022-02-09T09:14:38Z" level=info msg="getting the kong admin api client configuration" logger=setup
time="2022-02-09T09:14:38Z" level=error msg="tag filtering disabled because Kong Admin API does not support tags" error="HTTP status 401 (message: \"Invalid RBAC credentials\")" logger=setup
Error: could not retrieve Kong admin root: HTTP status 401 (message: "Invalid RBAC credentials")
Error: could not retrieve Kong admin root: HTTP status 401 (message: "Invalid RBAC credentials")

kong 版本2.7.0.0-alpine
k8s 版本v1.22.4

**Comment by raoxiaoyan** [comment, marked as answer]
Could talk to me in more detail.
“Invalid RBAC credentials” will happen:
1. admin hasn't assigned any workspace;
2. the user is a guest login in the system;
3. maybe you did not use the new token for the kong config. 
Please confirm it. Thanks.

**Comment by raoxiaoyan** [comment]
I can reproduce the problem.

1.reset RBAC token on the profile page. It’s will return a new token as the following figure.
![image](https://user-images.githubusercontent.com/22835296/159830997-5c2dad38-e746-4655-8306-9c264a246b82.png)
2. use ‘curl http://localhost:8001/ -H 'Kong-Admin-Token:oldpassword'', an error msg 'Invalid RBAC credentials’ will happen.

![image](https://user-images.githubusercontent.com/22835296/159831033-2f6a2d25-be2b-4657-830e-8349f4924671.png)

3. use new token access to 'http://localhost/:8001/' what we are expected. 

“curl http://localhost:8001/ -H ‘Kong-Admin-Token:VpYNoJBSnTWzbHXbqG9B1SgNIPe6wcPW’
![image](https://user-images.githubusercontent.com/22835296/159831063-d910e01b-ebdb-4acf-9ddd-4fbc8769144e.png)
